tensorflow/compiler/xla/service/gpu/gpu_compiler.cc

```diff
@@ -165,6 +165,12 @@ Status GpuCompiler::OptimizeHloModule(
       // where possible.  Not every batchnorm op can be implemented as a call to
       // cudnn, so decompose any remaining batchnorm ops into a soup of HLOs.
       if (hlo_module->config().debug_options().xla_gpu_use_cudnn_batchnorm()) {
+        // Since BatchNorm inference is essentially pointwise operations, it is
```

Comment:
It isn't just a performance consideration right, it looks like this expansion step is also working around some issue in `CudnnBatchNormRewriter`?

Reply:
Technically yes...`CudnnBatchNormRewriter` creates input descs by collapsing all physical dimensions "slower" than the channel dimension to  dim[0] and creating an `NCHW` descriptor by default. Although that is not incorrect, cudnn has restrictions on dim[0] of input desc for batchnorm inference. This will work around that restriction/unsupported configuration.